system/jlib/jptree.cpp

```diff
@@ -531,6 +541,340 @@ const char *queryHead(const char *xpath, StringBuffer &head)
 
 ///////////////////
 
+static constexpr unsigned defaultSiblingMapThreshold = 100;
+static unsigned siblingMapThreshold = (unsigned)-1; // off until configuration default it on.
```

Comment:
Not sure what best to do by default.
Testing suggest even small sets (10) benefit from a map, if looked up more than a few times.
However, it does add extra memory.

As it stands, it is off until if/the configuration is loaded, at which point it defaults to 100.